fix(insights): Show group analytics in exported/shared insights

Twixes opened the pull request:
## Problem

Resolves #17033.

## Changes

We were trying to fetch group types from the API, but the group types endpoint didn't support `sharing_access_token`. Now it does.

Also fixed the weird lack of a check mark (seen in the original issue's screenshot), and gave precedence to sharing access token authentication, so that shared assets work also when already logged into PostHog but with an account that doesn't normally have access to the asset.

Twixes committed:
fix(insights): Show group analytics in exported/shared insights

Twixes committed:
Also fix empty check mark

Twixes committed:
Mock group types response in stories

Twixes committed:
Update test assertions

Twixes committed:
Also update frontend logic tests

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)

Twixes committed:
Update propertyDefinitionsTableLogic.test.ts

benjackwhite reviewed (approved):
Nothing stands out as wrong and assuming the tests will catch anything if so :D Nice tidying here!

Twixes committed:
Fix an issue in group types viewset and add tests for good measure

More context here: https://posthog.slack.com/archives/C0428JHABK2/p1692361040699229

Twixes committed:
Merge branch 'master' into fix-group-analytics-shared

github-actions[bot] committed:
Update query snapshots

github-actions[bot] committed:
Update UI snapshots for `chromium` (2)
